### PR TITLE
@loopback/cli: Add findOne method to controller rest generator #5531

### PR DESCRIPTION
**Pull request by mrnetforge**
Implements #5531

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [ ] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

**Comment by raymondfeng** [comment]
@mrnetforge You don't have to manually fix snapshots. Run the test with `UPDATE_SNAPSHOTS=1` env var to automatically update the snapshots.

**Comment by agnes512** [comment]
@mrnetforge any updates? 😄

File: docs/site/Controller-generator.md [hunk 1]
```diff
@@ -216,6 +216,24 @@ export class TodoController {
...
+            schema: getModelSchemaRef(Todo, {includeRelations: true}),
+          },
+        },
+      },
+    },
+  })
+  async findOne(
+    @param.where(Todo) where?: Where<Todo>
+  ): Promise<Todo | null> {
+    return this.todoRepository.findOne(where);
```
Comment (bajtos) [inline docs/site/Controller-generator.md:234]: What is the HTTP response when no model was found? Do we return `200 OK` and `null` in the response body? Should we return `404 Not Found` instead?

I feel it's important to have a test for this edge case. Since we don't run HTTP tests for controllers scaffolded by CLI, it may be easier to add `findOne` to [`TodoController`](https://github.com/strongloop/loopback-next/blob/85a616415253bdca0475b7db9072da1ee611ac28/examples/todo/src/controllers/todo.controller.ts)  and then add the test to [examples/todo/src/__tests__/acceptance/todo.acceptance.ts](https://github.com/strongloop/loopback-next/blob/85a616415253bdca0475b7db9072da1ee611ac28/examples/todo/src/__tests__/acceptance/todo.acceptance.ts).

**Comment by dhmlau** [comment]
We just switch the contribution method from CLA to DCO, making your contribution easier in the future. Please sign the commits with DCO by amending your commit messages with `-s` flag and push the changes again. If you're using the command line, you can:

```
git commit --amend -s
git push --force-with-lease
```

Please refer to [this docs page](https://loopback.io/doc/en/contrib/code-contrib.html#developer-certificate-of-origin-dco) for details. Thanks!
